sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart

```diff
@@ -1,20 +1,101 @@
 import 'package:flutter/material.dart';
+// for Dart API calls
+import 'package:http/http.dart' as http; // use API calls (GET...)
+import 'dart:convert'; // convert GET result to JSON/List
+// for local file storage
+import 'package:path_provider/path_provider.dart'; // use getApplicationDocumentsDirectory()
+import 'dart:io'; // use Directory, File...
 
 /* global variable for initial food list for all new users.
-  will be filled as a future/from API call.
-   Out Grocery list. */
-final _groceryItems = ['Eggs', 'Milk', 'Fish Sauce', 'Bread', 'Apple Juice', 'Coke', 'Potato Chips',
-  'Froot Loops', 'Oatmeal', 'Snickers', 'Cheddar Cheese', 'Beer', 'Water', 'Cigarettes',
-  'Oreos', 'Ham', 'Wine', 'Bananas', 'Spinach', 'Cherries', 'Ketchup', 'Coffee', 'Hot Dogs'];
+  will be filled as a future/from API call */
+final _groceryItems = [];
 
 // used for sorting based on frequency
 final _groceryItemsFrequency = [];
 
-void main() {
+/* used for unsorting: maintain the old order. Also for frequency sort.
+  can also be used to keep old removed items, and add them back */
+var _groceryItemsCopy = [];
+
+var loggedIn = false; // needs to check if the user is logged in, currently does nothing
+
+var filePath = ''; // path to where this app saves files, calculated in main()
+
+// stores if the list has been alphabetically sorted. used if a new item is added, to put in proper place
+var _alphabeticallySorted = false;
+var _frequencySorted = false;
+
+// make main() async for Futures
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  // get the phone's path to where it saves files for this app
+  Directory appDocumentsDirectory = await getApplicationDocumentsDirectory();
+  var appDocumentsPath = appDocumentsDirectory.path;
+  filePath = "$appDocumentsPath";
+
+  // loggedIn boolean NOT IMPLEMENTED YET
+  if (loggedIn) {
+    /** get groceryList from API, check what group they're in...
+        ACTUALLY, don't we want each user to have their own local list?
+        SO, we probably don't want this.
```

Comment:
Correct!

Reply:
We'll probably still need a loggedIn bool for the shared part of the list later on, once we get to where we can implement it